utils/staking-miner/src/monitor.rs

```diff
@@ -86,6 +86,9 @@ macro_rules! monitor_cmd_for { ($runtime:tt) => { paste::paste! {
 			let hash = now.hash();
 			log::debug!(target: LOG_TARGET, "new event at #{:?} ({:?})", now.number, hash);
 
+			// if the runtime version has changed, terminate
+			crate::check_versions::<Runtime>(client, false).await?;
```

Comment:
Not neccessary for this PR, but is there a way we could parrell-ize the calls to `check_versions`, `ensure_signed_phase` & `create_election_ext`? Basicaly a Promise.all and then we can analyze the results of each independtly and figure out how to proceed

Reply:
yeah this is a good improvement. Also a critical improvement. is that currently we need to scrape the data of the whole pallet to check if we have a solution in this round or not (since we pass the entire `ext`), this can be done much better by just querying the appropriate storage item (see `ensure_no_previous_solution`). 

we should be able to early-exit from this check, without the need to scrape the entire `ext`.

Reply:
https://github.com/paritytech/polkadot/issues/3646